docs/src/main/asciidoc/stork-kubernetes.adoc

```diff
@@ -0,0 +1,479 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Getting Started with SmallRye Stork
+:extension-status: preview
+
+// Temporary until stork in in the BOM
+:stork-version: 1.1.0
+include::./attributes.adoc[]
+
+The essence of distributed systems resides in the interaction between services.
+In modern architecture, you often have multiple instances of your service to share the load or improve the resilience by redundancy.
+But how do you select the best instance of your service?
+That's where https://smallrye.io/smallrye-stork[SmallRye Stork] helps.
+Stork is going to choose the most appropriate instance.
+It offers:
+
+* Extensible service discovery mechanisms
+* Built-in support for Consul and Kubernetes
+* Customizable client load-balancing strategies
+
+include::./status-include.adoc[]
+
+== Prerequisites
+
+:prerequisites-docker:
+include::includes/devtools/prerequisites.adoc[]
+* Access to a Kubernetes cluster (Minikube is a viable option)
+
+== Architecture
+
+In this guide, we will work with a few components deployed in a Kubernetes cluster:
+
+* A simple blue service.
+* A simple red service.
+* The `color-service` is the Kubernetes service which is the entry point to the Blue and Red instances.
+* A REST Client calling the blue or red service (the selection is delegated to Stork). The REST Client is also deployed in a namespace within a Kubernetes cluster
+* A REST endpoint using the REST client and calling the services
+
+image::stork-kubernetes-architecture.png[Architecture of the application,width=100%, align=center]
+
+For the sake of simplicity, everything will be deployed in the same namespace of the Kubernetes cluster.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the applications step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `stork-kubernetes-quickstart` {quickstarts-tree-url}/stork-kubernetes-quickstart[directory].
+
+== Discovery and selection
+
+Before going further, we need to discuss discovery vs. selection.
+
+- Service discovery is the process of locating service instances.
+It produces a list of service instances that is potentially empty (if no service matches the request) or contains multiple service instances.
```

Comment:
```suggestion
It produces a list of service instances that is potentially empty (if no service matches the request).
```

if we leave `or contains multiple service instances`, we should probably also add that it can contain a single one too ;)